Improve error handling and response parser (#232)
MyUnisoft/httpie: docs/errors.md

# ERRORS

All errors generated by Httpie failure inherit [`HttpieError`](../src/class/HttpieCommonError.ts).

```ts
interface HttpieError {
  headers: IncomingHttpHeaders;
  statusCode: number;
}
```

## Tools

### isHttpieError

> The `isHttpieError` function can be used to find out if the error is an Httpie error. If the result is `false`, it is probably an error by Undici.

Reviewer: ```suggestion
The `isHttpieError` function can be used to find out weither the error is a `@myunisoft/httpie` or a `undici` error.
```
wdyt ?

```ts
import * as httpie from "@myunisoft/httpie";

try {
  await httpie.request("GET", "127.0.0.1");
}
catch (error) {
  if (httpie.isHttpieError(error)) {
    // This error inherits from HttpieError.
    console.log(Boolean(error.headers)) // true
    console.log(Boolean(error.statusCode)) // true
  }
  else {
    // This error can be of any type of error.
    console.log(Boolean(error.headers)) // false
    console.log(Boolean(error.statusCode)) // false
  }
}
```

### isHTTPError

The `isHTTPError` function can be used to find out if it is an HTTP error.
```ts
function isHTTPError(error: any): boolean;
```

Example:
```ts
import * as httpie from "@myunisoft/httpie";

try {
  await httpie.request("GET", "127.0.0.1");
}
catch (error) {
  if (httpie.isHTTPError(error)) {
    console.log(Boolean(error.data)) // true
    console.log(Boolean(error.statusMessage)) // true
  }
  else {
    console.log(Boolean(error.data)) // false
    console.log(Boolean(error.statusMessage)) // false
  }
}
```

---

## HTTP errors 

If the `RequestOptions.throwOnHttpError` option is set to true, all HTTP responses with a status code higher than 400 will generate an `HttpieOnHttpError` error.

> > [NOTE] Use [`isHTTPError`](#ishttperror) function to know if it is an HTTP error.

Reviewer: ```suggestion
> [!NOTE] 
> Use [`isHTTPError`](#ishttperror) function to know if it is an HTTP error.
```
